bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemStateConditionHandler.java

```diff
@@ -93,122 +92,99 @@ public boolean isSatisfied(Map<String, Object> inputs) {
+            } else {
+                logger.warn("Condition '{}' cannot be compared to the incompatible state '{}' from item {}.", state,
+                        qtState, itemName);
+            }
+        } else if (itemState instanceof DecimalType && null != compareState) {
+            DecimalType decimalState = compareState.as(DecimalType.class);
+            if (null != decimalState) {
+                return ((DecimalType) itemState).compareTo(decimalState) <= 0;
+            }
+        }
+        return false;
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked", "null" })
+    private boolean greaterThanOrEqualsToItemState(String itemName, String state) throws ItemNotFoundException {
+        Item item = itemRegistry.getItem(itemName);
+        State compareState = TypeParser.parseState(item.getAcceptedDataTypes(), state);
+        State itemState = item.getState();
+        if (itemState instanceof QuantityType) {
+            QuantityType qtState = (QuantityType) itemState;
+            if (compareState instanceof DecimalType) {
+                // allow compareState without unit -> implicitly assume its the same as the one from the
+                // state, but warn the user
+                logger.warn(
+                        "Received a QuantityType state '{}' with unit for item {}, but the condition is defined as a plain number without unit ({}), please consider adding a unit to the condition.",
+                        qtState, itemName, state);
+                return qtState.compareTo(
+                        new QuantityType<>(((DecimalType) compareState).toBigDecimal(), qtState.getUnit())) >= 0;
+            } else if (compareState instanceof QuantityType) {
+                return qtState.compareTo((QuantityType) compareState) >= 0;
+            } else {
+                logger.warn("Condition '{}' cannot be compared to the incompatible state '{}' from item {}.", state,
+                        qtState, itemName);
+            }
+        } else if (itemState instanceof DecimalType && null != compareState) {
+            DecimalType decimalState = compareState.as(DecimalType.class);
+            if (null != decimalState) {
+                return ((DecimalType) itemState).compareTo(decimalState) >= 0;
+            }
+        }
+        return false;
+    }
+
+    @SuppressWarnings("null")
+    private boolean equalsToItemState(String itemName, String state) throws ItemNotFoundException {
+        Item item = itemRegistry.getItem(itemName);
+        State compareState = TypeParser.parseState(item.getAcceptedDataTypes(), state);
+        State itemState = item.getState();
+        if (itemState instanceof QuantityType && compareState instanceof DecimalType) {
+            logger.warn(
+                    "Received a QuantityType state '{}' with unit for item {}, but the condition is defined as a plain number without unit ({}), please consider adding a unit to the condition.",
```

Comment:
"please consider adding" sounds as if it is optional and will work nonetheless without.
Shouldn't we maybe use some stronger formulation to make it clear that the current configuration is invalid and the comparison is simply not done (always returns false)?

Reply:
I reformulated the warning to be more precise.

Additionally I added a check to deal with dimensionless `QuantityType`s without unit (internally `AbstractUnit.ONE`) because for them it is not possible to define a unit in the compare condition as the unit symbol is an empty string.